runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -618,6 +638,113 @@ else if (event.getMessage().startsWith(KILLED_TELEBLOCK_OPPONENT_TEXT))
 				}
 			}
 		}
+
+		if (config.showTzhaarTimers())
+		{
+			String message = Text.removeTags(event.getMessage());
+			Matcher matcher = TZHAAR_COMPLETE_MESSAGE.matcher(message);
+
+			if (message.contains(TZHAAR_DEFEATED_MESSAGE) || matcher.matches())
+			{
+				removeTzhaarTimer();
+				resetTzhaarConfig();
+				resetTzhaarVars();
+				return;
+			}
+
+			Instant now = Instant.now();
+			matcher = TZHAAR_PAUSED_MESSAGE.matcher(message);
+			if (matcher.matches())
+			{
+				tzhaarLastTime = now;
+				createTzhaarTimer(tzhaarStartTime, now);
+				return;
+			}
+
+			matcher = TZHAAR_WAVE_MESSAGE.matcher(message);
+			if (!matcher.matches())
+			{
+				return;
+			}
+
+			if (!tzhaarStarted)
+			{
+				int wave = Integer.parseInt(matcher.group(1));
+				if (wave != 1)
+				{
+					return;
+				}
+
+				tzhaarStarted = true;
+				tzhaarStartTime = now;
+			}
+
+			else if (tzhaarLastTime != null)
+			{
+				tzhaarStartTime = tzhaarStartTime.plus(Duration.between(tzhaarStartTime, now)).minus(Duration.between(tzhaarStartTime, tzhaarLastTime));
+				tzhaarLastTime = null;
+			}
+
+			createTzhaarTimer(tzhaarStartTime, tzhaarLastTime);
+		}
+
+	}
+
+	private void updateInfoBoxState()
```

Comment:
This isn't really what this method is doing. It's possibly removing the timer and resetting config/vars. It should be named appropriately.